### PR TITLE
Update Helloworld unit test error message to make it actionable

### PR DESCRIPTION
**Pull request by chaodaiG**
Update error message with detailed explanation so that developers know how to debug if unit test ever failed.
Fix a bug in unit test.
Update all READMEs so that they are valid

Part of: #1793

### CODE REVIEW

File: docs/serving/samples/hello-world/helloworld-nodejs/index.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  const target = process.env.TARGET || 'World';
+  const target = process.env.TARGET || "World";
   res.send(`Hello ${target}!`);
 });
 
 const port = process.env.PORT || 8080;
 app.listen(port, () => {
-  console.log('Hello world listening on port', port);
+  console.log("Hello world listening on port", port);
 });
```
Comment (grayside) [inline docs/serving/samples/hello-world/helloworld-nodejs/index.js]: All the changes in this file should be reverted.

File: test/sampleapp/config.yaml [hunk 2]
```diff
@@ -43,10 +45,12 @@ languages:
       - "Dockerfile"
   - language: "ruby"
     expectedOutput: "Hello Ruby Sample v1!"
+    preCommands:
+      - exec: "cp"
+        args: "../../docs/serving/samples/hello-world/helloworld-ruby/Gemfile.lock helloworld-ruby_tmp/Gemfile.lock"
```
Comment (grayside) [inline test/sampleapp/config.yaml]: If the intention is to have custom cp statements for every file which is not inlined, there is more, for example, node.js has `package-lock.json`. A language-by-language review will be necessary.

File: test/sampleconsistency/sampleconsistency_test.go [hunk 3]
```diff
@@ -85,8 +88,20 @@ func checkContains(t *testing.T, rl []string, src string) {
 		}
 		ir++
 	}
-	if is < len(sl) && best < len(sl) && best != -1 {
-		t.Fatalf("README.md file is missing line %d ('%s') from file '%s'", best, sl[best], src)
+
+	actionMsg := "All files required for running Helloworld sample apps are checked " +
```
Comment (grayside) [inline test/sampleconsistency/sampleconsistency_test.go]: Why is this limited to HelloWorld? All the knative samples are required to have the synchronization between code and README.

File: test/sampleconsistency/sampleconsistency_test.go [hunk 4]
```diff
@@ -85,8 +88,20 @@ func checkContains(t *testing.T, rl []string, src string) {
...
+		"against README.md, the content of source files should be identical with what's " +
+		"in README.md file, the list of the files to be verified is the same set of files " +
+		"used for running sample apps, they are configured in `/test/sampleapp/config.yaml`. " +
+		"If an exception is needed the file can be configured to be copied as a separate step " +
+		"in `PreCommand` such as: " +
+		"https://github.com/knative/docs/blob/65f7b402fee7f94dfbd9e4512ef3beed7b85de66/test/sampleapp/config.yaml#L4"
```
Comment (grayside) [inline test/sampleconsistency/sampleconsistency_test.go]: Seems like there should be a comment referencing this message from the current config and/or file copying implementation to ensure changes are followed through to this message.

File: docs/serving/samples/hello-world/helloworld-nodejs/index.js (hunk 1 above)
Comment (chaodaiG) [reply docs/serving/samples/hello-world/helloworld-nodejs/index.js]: done

File: test/sampleapp/config.yaml (hunk 2 above)
Comment (chaodaiG) [reply test/sampleapp/config.yaml]: done

Comment (chaodaiG) [reply test/sampleapp/config.yaml]: Did a language-by-language review and that's already the last piece missing

File: test/sampleconsistency/sampleconsistency_test.go (hunk 3 above)
Comment (chaodaiG) [reply test/sampleconsistency/sampleconsistency_test.go]: Removed `HelloWorld` from the error message, will let docs team triage the importance of other sample apps
@samodell

File: test/sampleconsistency/sampleconsistency_test.go (hunk 4 above)
Comment (chaodaiG) [reply test/sampleconsistency/sampleconsistency_test.go]: Moved the message to where the config and copy was defined and reference it from unit test

**Comment by chaodaiG** [comment]
@grayside , thanks for reviewing, please take another look

**Pull request merged**